cds_migrator_kit/rdm/migration/load/load.py

```diff
@@ -172,12 +173,32 @@ def _after_publish_update_dois(self, identity, record, entry):
                 _draft = current_rdm_records_service.edit(identity, record["id"])
                 current_rdm_records_service.publish(identity, _draft["id"])
 
-    def _after_publish_update_created(self, record, entry):
-        """Update created timestamp post publish."""
-        # Fix the `created` timestamp forcing the one from the legacy system
-        # Force the created date. This can be done after publish as the service
-        # overrides the `created` date otherwise.
-        record._record.model.created = arrow.get(entry["record"]["created"]).datetime
+    def _after_publish_update_created(self, record, entry, version):
+        """Update created timestamp post publish.
+
+        Ensures that the `created` timestamp is correctly set, preferring:
+        1. The original legacy system value.
+        2. The file's creation date if the original value is invalid.
+        3. Today's date if the original value is missing.
+        """
+        creation_date = entry.get("record", {}).get("created")
+        is_created_today = datetime.date.today().isoformat() == creation_date
```

Comment:
I would prefer we catch it before the record is even loaded. In the transform phase, it will avoid bumping revisions of the records only due to migration mistakes

Reply:
Not sure I understand well this part. We can definitely chek in the transform but we cannot set it, unless I am missing something

Reply:
Discussed IRL, we can rely on the version creation_date